indico/modules/users/models/users.py

```diff
@@ -122,6 +140,20 @@ def full_name(self):
 }
 
 
+def format_display_full_name(user, obj):
+    name_format = user.settings.get('name_format') if user else NameFormat.first_last
+    upper = name_format in (NameFormat.first_last_upper, NameFormat.f_last_upper, NameFormat.last_f_upper,
+                            NameFormat.last_first_upper)
+    if name_format in (NameFormat.first_last, NameFormat.first_last_upper):
+        return obj.get_full_name(last_name_first=False, last_name_upper=upper, abbrev_first_name=False)
+    elif name_format in (NameFormat.last_first, NameFormat.last_first_upper):
+        return obj.get_full_name(last_name_first=True, last_name_upper=upper, abbrev_first_name=False)
+    elif name_format in (NameFormat.last_f, NameFormat.last_f_upper):
+        return obj.get_full_name(last_name_first=True, last_name_upper=upper, abbrev_first_name=True)
+    elif name_format in (NameFormat.f_last, NameFormat.f_last_upper):
+        return obj.get_full_name(last_name_first=False, last_name_upper=upper, abbrev_first_name=True)
```

Comment:
```
else:
    raise ValueError('Invalid name format: {}'.format(name_format))
```

Otherwise you'd just return `None` in such a case (which should never happen anyway)